Data/core_rulebook/cr_abilities.lua

```diff
@@ -0,0 +1,10427 @@
+      },
+      Conditions={
+        function (character)
+          return (character.Variables["InterativeAttack"] == 1)
+        end,
+      },
+    },
+    {
+      Name="NaturalAttackToHit",
+      FormatString="+%1/+%2",
+      Arguments={
+        "ShieldBashAttackTotal",
+        "ShieldBashAttackTotal-5",
+      },
+      Conditions={
+        function (character)
+          return (character.Variables["InterativeAttack"] == 2)
+        end,
+      },
+    },
+    {
+      Name="NaturalAttackToHit",
+      FormatString="+%1/+%2/+%3",
+      Arguments={
+        "ShieldBashAttackTotal",
+        "ShieldBashAttackTotal-5",
+        "ShieldBashAttackTotal-10",
+      },
+      Conditions={
+        function (character)
+          return (character.Variables["InterativeAttack"] == 3)
+        end,
+      },
+    },
+    {
+      Name="NaturalAttackToHit",
+      FormatString="+%1/+%2/+%3/+%4",
+      Arguments={
+        "ShieldBashAttackTotal",
+        "ShieldBashAttackTotal-5",
+        "ShieldBashAttackTotal-10",
+        "ShieldBashAttackTotal-15",
+      },
+      Conditions={
+        function (character)
+          return (character.Variables["InterativeAttack"] == 4)
+        end,
+      },
+    },
+    {
+      Name="NaturalAttackDamage",
+      FormatString="%1d%2",
+      Arguments={
+        "ShieldDamageDice",
+        "ShieldDamageSize",
+      },
+    },
+    {
+      Name="NaturalAttackDamageBonus",
+      FormatString="%1",
+      Arguments={
+        "STR+ShieldBashDamageBonus",
+      },
+    },
+    {
+      Name="NaturalAttackReach",
+      FormatString="%1 ft.",
+      Arguments={
+        "NaturalShieldReach",
+      },
+    },
+    {
+      Name="NaturalAttackType",
+      FormatString="Lethal",
+    },
+    {
+      Name="NaturalAttackThreatRange",
+      FormatString="%1/x%2",
+      Arguments={
+        "ShieldThreatRange",
+        "ShieldCritMult",
+      },
+    },
+  },
+})
+
+DefineAbility({
+  Name="Shield Bash Stats",
+  Category="Internal",
+  AllowMultiple=false,
+  Stackable=false,
+  Bonuses={
+    {
+      Category="VAR",
+      Variables={
+        "ShieldDamageDice",
+      },
+      Formula="1",
+      Type={
+        Name="Base",
+        Replace=false,
+        Stack=false,
+      },
+    },
+    {
+      Category="VAR",
+      Variables={
+        "ShieldBashAttackBonus",
+      },
+      Formula="-4",
+      Conditions={
+        function (character)
+          return not (1 <= #filter(character.Abilities, function (ability)
+            return ability.Category == "FEAT" and (ability.Name == "Shield Mastery" or ability.Name == "Shield Master")
+          end))
+        end,
+      },
+    },
+  },
+})
+
+ModifyAbility({
+  Name="Shield Bash Stats",
+  Category="Internal",
+  AllowMultiple=false,
+  Stackable=false,
+  Bonuses={
+    {
+      Category="VAR",
+      Variables={
+        "ShieldBashingDieSizeStep",
+      },
+      Formula="3",
+      Conditions={
+        function (character)
```

Comment:
This is _really_ verbose way of saying "the character is small, and they have a light shield", yes?  Is there any way to condense it at all?

Reply:
Honestly... this seems like this should just be a weapon, that would solve it right quick. :-) But I guess that's not how they implemented it *sigh*

Reply:
Yea, this is very weird. this function is a `PREMULT:2,[PRESIZE:M],[PREEQUIP:Type=Shield.Light]`, which is basically a very verbose "and".
This is something we can translate after getting everything setup.

Reply:
And... FFS, I think thats another bug, I think that looks for something equipped that is both "Type=Shield" and "Type=Light", so I need to parse out the period separated types.....